### PR TITLE
TicketBatch docs

### LINKED ISSUE #898: TicketBatch docs
Spec: https://github.com/xrp-community/standards-drafts/issues/16
PR: https://github.com/ripple/rippled/pull/3271

Docs Scope:

- Changes to sequence numbers (seq `0` when tickets are used) in probably a few places or mentions of sequence numbers. For example, "every transaction uses a sequence number" (or a ticket) and "every transaction increases its sender's sequence number by 1" (or sets aside multiple sequence numbers for tickets)
- New `TicketSequence` common transaction field
- New `TicketCreate` transaction type
- New `Ticket` ledger object type
- TBD: New serialization fields, error codes?
- Concept page describing the feature

### PR DESCRIPTION
**Pull request by mDuo13**
Fixes #898.

Documents all the new Ticket features introduced the TicketBatch amendment:

- [Concept page introducing Tickets](http://mduo13.github.io/xrpl-dev-portal/tickets.html), including a couple diagrams.
- Updated Known Amendments with `TicketBatch` replacing the `Tickets` amendment. Linked [the spec](https://github.com/xrp-community/standards-drafts/issues/16).
- Transaction reference updates: new [TicketCreate transaction type](http://mduo13.github.io/xrpl-dev-portal/ticketcreate.html), [`TicketSequence` common field](http://mduo13.github.io/xrpl-dev-portal/transaction-common-fields.html)
    - In the code, `TicketSequence` isn't a "common" field but is independently added to every transaction type except pseudo-transactions. From a user perspective, this is functionally identical to being a common field.
- [`Ticket` ledger object type reference](http://mduo13.github.io/xrpl-dev-portal/ticket.html).
- Updated API docs for `account_objects` and `ledger_entry` for fetching Tickets (was partially done already).
- Updated ledger types and transactions where the Ticket sequence is used instead of the transaction sequence to identify a specific object.
    - Note: This applies to OfferCreate's `OfferSequence` field for canceling a previous offer too, but the docs for OfferCreate just say, "in the same way as OfferCancel" so I left them unchanged.
    - Note 2: No change to CheckCancel or the PaymentChannelFund/Claim transaction docs because those use the ID rather than the sequence number. (These changed indirectly because the ID uses the sequence number, but that doesn't affect how users construct the transactions.)

Not in-scope for this PR:

- Tutorial for how to create & use a Ticket (to come in a later PR)

### CODE REVIEW

File: content/_img-sources/ticket-creation.uxf [hunk 1]
```diff
@@ -0,0 +1,208 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (mDuo13) [inline content/_img-sources/ticket-creation.uxf:1]: These `.uxf` files are the UMLet source files for the diagrams. You don't have to review them directly (they are exported as `.svg` files that are also committed to the repo)

File: content/concepts/payment-system-basics/accounts/tickets.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+However, there are some situations where sequence numbers are too limiting. For example:
+
+- Two or more users share access to an account, each with the ability to send transactions independently. If these users try to send transactions around the same time without coordinating first, they may each try to use the same Sequence number for different transactions, and only one can succeed.
+- You may want to prepare and sign a transaction in advance, then save it in some secure storage so that it can be executed at any future point if certain events occur. However, if you want to continue using your account as normal in the meantime, you don't know what Sequence number the set-aside transaction will need.
+- When [multiple people must sign a transaction](multi-signing.html) to make it valid, it can be difficult to plan more than one transaction at a time. If you number the transactions with separate sequence numbers, you can't send the later-numbered transactions until everyone has signed the previous transactions; but if you use the same sequence number for each pending transactions, only one of them can succeed.
```
Comment (scottschurr) [inline content/concepts/payment-system-basics/accounts/tickets.md:24]: Nice motivating examples!

File: content/concepts/payment-system-basics/accounts/tickets.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+To look up what Tickets an account has available, use the [account_objects method][].
+
+## Limitations
+
+Any account can create and use Tickets on any type of transaction. However, some restrictions apply:
+
+- Each Ticket can only be used once. It is possible to have multiple different candidate transactions
+- Each account cannot have more than 250 Tickets in the ledger at a time. Therefore, you cannot create more than 250 Tickets at a time, either.
+- You _can_ use a Ticket to create more Tickets. If you do, the Ticket you used does not count towards the total number of Tickets you can have at once.
+- Each Ticket counts toward the [owner reserve](reserves.html), so you must set aside 5 XRP for each Ticket you have not used yet. The XRP becomes available for you to use again after the Ticket is used.
```
Comment (scottschurr) [inline content/concepts/payment-system-basics/accounts/tickets.md:56]: Maybe this is obvious, but I wonder if it is worth mentioning...

> The transaction that creates tickets either succeeds and makes all of the requested tickets or it fails and creates none of them.  This is an important consideration if your account may not contain enough XRP to cover the reserve for all the tickets requested by the transaction.

Just a thought.  Your call.

**Review by scottschurr — APPROVED** [review]
:+1: Looks great!  As I read through this I was a bit surprised by how reasonable the rules sounded.  Thanks for all your help deciding what those rules should be, in addition to documenting them.

File: content/concepts/payment-system-basics/accounts/tickets.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+As with sequence numbers, sending a transaction uses up the Ticket _if and only if_ the transaction is confirmed by [consensus](consensus.html). However, transactions that fail to do what they were intended to do can still be confirmed by consensus with [`tec`-class result codes](tec-codes.html).
+
+To look up what Tickets an account has available, use the [account_objects method][].
+
+## Limitations
+
+Any account can create and use Tickets on any type of transaction. However, some restrictions apply:
+
+- Each Ticket can only be used once. It is possible to have multiple different candidate transactions
```
Comment (gbarr01) [inline content/concepts/payment-system-basics/accounts/tickets.md]: Missing a period after 

> It is possible to have multiple different candidate transactions

File: content/references/rippled-api/transaction-formats/transaction-types/ticketcreate.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}
+```
+
+{% include '_snippets/tx-fields-intro.md' %}
+<!--{# fix md highlighting_ #}-->
+
+| Field            | JSON Type        | [Internal Type][] | Description        |
+|:-----------------|:-----------------|:------------------|:-------------------|
+| `TicketCount`    | Number           | UInt32            | How many Tickets to create. This must be a positive number, and cannot cause the account to own more than 250 Tickets after executing this transaction. |
+
```
Comment (gbarr01) [inline content/references/rippled-api/transaction-formats/transaction-types/ticketcreate.md:39]: Not sure you need a comma here as the appositive is restrictive.

**Review by gbarr01 — APPROVED** [review]
LGTM

File: content/concepts/payment-system-basics/accounts/tickets.md (hunk 2 above)
Comment (mDuo13) [reply content/concepts/payment-system-basics/accounts/tickets.md:24]: They're basically just condensed versions of the ones from your spec, so credit goes to you more than me for that. ;)

File: content/concepts/payment-system-basics/accounts/tickets.md (hunk 3 above)
Comment (mDuo13) [reply content/concepts/payment-system-basics/accounts/tickets.md:56]: I added a note to this effect to the transaction description; I think details like this are more relevant there than on this concepts page.

File: content/concepts/payment-system-basics/accounts/tickets.md (hunk 4 above)
Comment (mDuo13) [reply content/concepts/payment-system-basics/accounts/tickets.md]: This was an incomplete sentence. The full version:

> It is possible to have multiple different candidate transactions that would use the same Ticket Sequence, but only one of those candidates can be validated by consensus.

**Review by SirJosh1987 — COMMENTED** [review]
One day GitHub will release my account hopefully. If someone would call me and explain what I can do to make this easier please do.

**Pull request merged**